changes/2024-09-13_cache-across-hierarchical-keyrings/change.md

```diff
@@ -0,0 +1,188 @@
+| .NET     | 1.7.0              | n/a             | [CreateAwsKmsHierarchicalKeyringInput.cs](https://github.com/aws/aws-cryptographic-material-providers-library/blob/main/AwsCryptographicMaterialProviders/runtimes/net/Generated/AwsCryptographicMaterialProviders/CreateAwsKmsHierarchicalKeyringInput.cs)                                         |
+
+## Definitions
+
+An "MPL Consumer" is a library, service, or other application
+that uses the AWS Cryptographic Material Providers Library (MPL)
+to manage cryptographic materials.
+An "MPL Consumer" MAY be using an AWS Crypto Tools product,
+such as the AWS Encryption SDK or AWS Database Encryption SDK.
+
+By "KMS Relationship", we mean any or all of the following:
+
+- KMS Configuration
+- Credentials used when creating the KMS Client, and thus used when calling KMS
+- Other properties of the KMS Client, such as the region, or request headers
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+To improve the MPL Consumer data key caching experience,
+when using the Hierarchical Keyring,
+we need to allow caching across Key Stores/KMS Clients/KMS Keys.
+
+To facilitate caching across Key Stores/KMS Clients/KMS Keys,
+we MUST break the Cryptographic Materials Cache (CMC)
+out of the Hierarchical Keyring.
+
+By allowing MPL Consumers to optionally provide an initialized shared CMC
+to the Hierarchical Keyring at construction,
+the CMC MAY cache Branch Keys protected by different
+KMS Relationships.
+
+This simplifies Multiple KMS Relationship MPL Consumers,
+as they do not need to stand up an LRU Cache of Hierarchical Keyrings.
+
+Instead, they may maintain one CMC.
+They still create a Hierarchical Keyring instance per KMS Relationship,
+and they MUST use the correct Keyring to retrieve material
+from the Cache.
+
+But they only need to maintain the common cache.
+
+In the future, the CachingCMM will be introduced to
+Crypto Tool's Dafny products.
+The CachingCMM and the Hierarchical Keyring both consume CMCs.
+Thus, it will be possible to provide a CMC to both a
+Hierarchical Keyring & a CachingCMM.
+This cache identifier design considers
+how the cache entries of multiple Hierarchical Keyrings
+and CachingCMMs will be appropriately separated.
+
+## Out of Scope
+
+- Supporting a shared Cache in the DB-ESDK to fetch
+  BeaconKeyMaterials across multiple Searchable
+  Encryption Configurations, or across multiple KMS Relationships.
+
+## Motivation
```

Comment:
feels like motivation should be before the summary

Reply:
If you read the whole doc then yes, you're right. But looking at the previous change docs, Summary is at the top because that is the most important thing anyone is going to read.